nexus/src/db/datastore/rack.rs

```diff
@@ -83,19 +85,21 @@ impl DataStore {
         opctx: &OpContext,
         rack_id: Uuid,
         services: Vec<Service>,
+        datasets: Vec<Dataset>,
     ) -> UpdateResult<Rack> {
```

Comment:
I think at one point in the RFD 278 discussion we talked about ensuring that if we got a second RSS initialization request, it must have the same parameters as the first one.  Is that checked somehow here or are we deferring that work for now?

Reply:
We aren't checking the parameters right now, but we do bail out early if we have ever completed a call to `rack_set_initialized`: 

https://github.com/oxidecomputer/omicron/blob/362f7e13ca360c240d49ee1019ca9d4435c9e6f6/nexus/src/db/datastore/rack.rs#L106-L120

This means that we won't ever try to create conflicting state, but rather enforcing a "first-one-wins" policy.